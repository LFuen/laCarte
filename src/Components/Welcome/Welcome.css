.welcomePage {
    margin-bottom: 100px;
}

button.shadow {
    color: black;
    border: 3px solid rgba(233, 111, 111);
    background-image: radial-gradient(rgb(255, 255, 255),  rgba(233, 111, 111));
    border-radius: 40px;
    padding: 15px 30px;
    text-decoration: none;
    text-shadow:2px 2px rgb(117, 117, 117);
    font-weight: bold;
    font-size: 18px;
    font-family: 'Special Elite', cursive;
}

button:hover,
button:focus {
    color:rgb(233, 111, 111);
    border: 3px solid rgb(0, 0, 0);
}

.titleImage{
    width: 30%;
}

.flexed {
    display: flex;
    justify-content: center;
    align-items: center;
}

.worked {
    font-family: 'Special Elite', cursive;
    font-size: 30px;
}

h5 {
    font-family: 'Special Elite', cursive;
    font-size: 20px;
}


/* 
=================================
    BUTTON EFFECT 
=================================
*/

.shadow
{
    position:relative;
    -webkit-box-shadow:0 10px 14px rgb(0, 0, 0), 0 0 40px rgba(0, 0, 0, 0.1) inset;
        -moz-box-shadow:0 10px 14px rgb(0, 0, 0), 0 0 40px rgba(0, 0, 0, 0.1) inset;
            box-shadow:0 10px 14px rgb(0, 0, 0), 0 0 40px rgba(0, 0, 0, 0.1) inset;
}
.shadow:before, .shadow:after
{
    content:"";
    position:absolute;
    z-index:-1;
    -webkit-box-shadow:0 0 20px rgb(0, 0, 0);
    -moz-box-shadow:0 0 20px rgb(0, 0, 0);
    box-shadow:0 0 20px rgb(0, 0, 0);
    top:0;
    bottom:0;
    left:10px;
    right:10px;
    -moz-border-radius:100px / 10px;
    border-radius:100px / 10px;
}
.shadow:after
{
    right:10px;
    left:auto;
    -webkit-transform:skew(8deg) rotate(3deg);
        -moz-transform:skew(8deg) rotate(3deg);
        -ms-transform:skew(8deg) rotate(3deg);
        -o-transform:skew(8deg) rotate(3deg);
            transform:skew(8deg) rotate(3deg);
}

/* 
=================================
    BUTTON EFFECT 
=================================
*/