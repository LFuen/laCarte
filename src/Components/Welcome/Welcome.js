import React, {Component} from 'react'
import './Welcome.css'



class Welcome extends Component {
    render() {
        return (
            <div className='welcomePage'>
                <span title='La Carte'><img className='titleImage' alt='LaCarte' src='https://lfuen.github.io/laCarte/images/laCarteFont.png'/></span> 
                <h3 className='worked'>The Healthy-Meal-Alternative App for the Busy Professional.</h3>
                <h5>(We didn't say workaholic, you did.)</h5>
                <div className='flexed'>
                    <span title='Bon Apetit!'><img className='mainImage' alt='laCarte' src='https://lfuen.github.io/laCarte/images/laCarte.png'/></span> 
                </div>
            </div>
        )
    }
}



export default Welcome