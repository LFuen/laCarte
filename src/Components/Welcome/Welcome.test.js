import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import LaCarteContext from '../../context/LaCarteContext'
import Welcome from './Welcome'



describe(`Welcome Component`, () => {
    describe(`Snapshot`, () => {
        it(`renders the UI as expected`, () => {
            const welcome = renderer.create(<Welcome/>)
                expect(welcome).toMatchSnapshot()
        })
    })

    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            
            const div = document.createElement('div')

            const value = {
                meals: 'string',
                chefs: 'string',
                cuisines: 'string',
                orders: 'string',
                addOrder: 'string',
                updateOrder: 'string',
                deleteOrder: 'string',
            };

            ReactDOM.render(
                    <LaCarteContext.Provider value={value}>
                        <Welcome />
                    </LaCarteContext.Provider>,
                div)
            ReactDOM.unmountComponentAtNode(div)
        })
    })
})