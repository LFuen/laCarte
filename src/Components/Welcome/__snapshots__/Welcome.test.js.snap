// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Component Snapshot renders the UI as expected 1`] = `
<div
  className="welcomePage"
>
  <span
    title="La Carte"
  >
    <img
      alt="LaCarte"
      className="titleImage"
      src="https://lfuen.github.io/laCarte/images/laCarteFont.png"
    />
  </span>
  <h3
    className="worked"
  >
    The Healthy-Meal-Alternative App for the Busy Professional.
  </h3>
  <h5>
    (We didn't say workaholic, you did.)
  </h5>
  <div
    className="flexed"
  >
    <span
      title="Bon Apetit!"
    >
      <img
        alt="laCarte"
        className="mainImage"
        src="https://lfuen.github.io/laCarte/images/laCarte.png"
      />
    </span>
  </div>
</div>
`;
